src/transformers/models/whisper/processing_whisper.py

```diff
@@ -91,3 +91,12 @@ def decode(self, *args, **kwargs):
         the docstring of this method for more information.
         """
         return self.tokenizer.decode(*args, **kwargs)
+
+    def get_prompt_ids(self, text: str):
+        """Converts prompt text to IDs that can be passed to [`~WhisperForConditionalGeneration.generate`]."""
+        prompt_token = "<|startofprev|>"
+        if prompt_token in text:
+            text = text.replace(prompt_token, "")
```

Comment:
Would it be better to strip out all special tokens / raise an error? 

Not sure what OpenAI does here. In "condition on previous text mode" they don't include the `<|startoftranscript|><|en|><|transcribe|><|notimestamps|>` tokens when they put the previous text in the `<|startofprev|>` section (since that would be problematic). But I'm not sure if they also strip out the actual timestamps such as `<|1.5|>` etc, that would be worth looking into.

I think `get_prompt_ids` should not accept any tokens `>= processor.tokenizer.eos_token`, so we should either strip these out or raise an error. If we do want to allow timestamp tokens, then it should accept tokens `> processor.tokenizer.all_special_ids[-1]`, since that's where the timestamp tokens begin.

Reply:
Looks like they have their tiktoken package handle it and it raises an error if any special token is included, so will look to do the same

Reply:
[These](https://github.com/openai/whisper/blob/c09a7ae299c4c34c5839a76380ae407e7d785914/whisper/tokenizer.py#L334-L345) are the tokens they strip out, so timestamps are included. `transformers` uses the same time_precision of 0.02, but notably even tho <|1.00|> is caught by OpenAI's special tokens check any number that doesn't have hundreths place precision like <|1.0|> isn't. I'd guess we still want to raise an error for these so I'll try to write a check for a token with any fp number inside

Reply:
Your solution is probably fine but a simpler approach would be to make sure no token has a value `>= processor.tokenizer.eos_token`. ;-)

Reply:
Oh interesting, for timestamps and <|nospeech|> too? I get several low ids when trying to tokenize a timestamp

![Screenshot 2023-05-04 at 9 35 47 AM](https://user-images.githubusercontent.com/78612354/236221427-ebd82ee5-3584-4422-b771-788ffedebea6.png)

Reply:
Maybe related: [#20225](https://github.com/huggingface/transformers/issues/20225#issuecomment-1534911055).

Reply:
I guess our tokenizer only decodes timestamp tokens, but doesn't know how to encode them?

Timestamps start at 50364. Any token id higher than that is a timestamp token.